lib/developer_portal/lib/liquid/forms/login.rb

```diff
@@ -13,6 +13,10 @@ def form_options
       def path
         session_path
       end
+
+      def recaptcha_action
+        DeveloperPortal::Engine.routes.recognize_path(login_path).fetch(:controller)
```

Comment:
this returns a string?

Reply:
btw can this be a universal solution for the base bot protected `Form` module? I mean in case we use `path` instead of `login_path`.

Reply:
yes:

```ruby
DeveloperPortal::Engine.routes.recognize_path(DeveloperPortal::Engine.routes.url_helpers.login_path)
=> {:controller=>"developer_portal/login", :action=>"new"}

```

Reply:
> btw can this be a universal solution for the base bot protected Form module? I mean in case we use path instead of login_path.

No, because `path` here is `'/session'`, which is different to `login_path` (`'developer_portal/login'`)

Also, the `Form` module is also included from `SemanticFormBuilder`, which requires a `recaptcha_action` equal to `controller.controller_path` because it doesn't have a `path` method defined.

Reply:
That's actually awesome. Can we add `action` as part of recaptcha `action` string too? Just in controller we will need to call recaptcha verify also with that added. Or idk, maybe for another life.

Reply:
> Also, the Form module is also included from SemanticFormBuilder, which requires a recaptcha_action equal to controller.controller_path because it doesn't have a path method defined.

How to we set form action in there then?

Reply:
You showed to me `{:controller=>"developer_portal/login", :action=>"new"}` so we have `new` by way of getting the controller from the path.

Reply:
For `BotProtected` forms is easy, because they always define `path` and `http_method`, which are the two parameters we need to get the route. But `SemanticFormBuilder` forms don't, so either we find a way to get the path and the method for the form action while in the `GET` request, or we assume the form `action` will always be `controller_path` (it's sending the request to the same controller that loads the form) and method will always be `POST`, thus invalidating all `PATCH` or `PUT` forms. I think it's too error prone, We are already assuming `SemantincFormBuilder` form are always sending request to the same controller, which is a bit risky.

Reply:
ok, we can think about it again if ever needed